src/hooks/useExpiryDateMask.js

```diff
@@ -12,49 +12,75 @@ export const useExpiryDateMask = initialState => {
+    7: /((^[012][1-9])|(^[3][01]))\/(([0][1-9])|([1][012]))\/\d{1}$/,
+    // Eight characters match:
+    // - a valid day and valid month and a two digit year.
+    8: /((^[012][1-9])|(^[3][01]))\/(([0][1-9])|([1][012]))\/\d{2}$/,
+    // Nine characters match:
+    // - a valid day and valid month and a three digit year.
+    9: /((^[012][1-9])|(^[3][01]))\/(([0][1-9])|([1][012]))\/\d{3}$/,
+    // Ten characters match:
+    // - a valid day and valid month and a three digit year.
+    10: /((^[012][1-9])|(^[3][01]))\/(([0][1-9])|([1][012]))\/\d{4}$/,
   };
 
   // Validates a new value being set as state. Either allows the new value, or rejects
   // it, setting the current value as state. Appends/prepends a 0 or / to values of
   // length two. Resets the state when backspacing into an inconsistent state. Only
-  // allows values in the form MM-Y[YYY]
+  // allows values in the form DD/MM/Y[YYY]
   const checkAndSetDate = value => {
     const { length: oldLength } = date;
     const { length: newLength } = value;
 
-    // Helpers - prepend/append / or 0 for values in M/ or MM state.
-    const adjustMonth = newValue => (newValue.endsWith('/') ? `0${newValue}` : `${newValue}/`);
+    // Helpers - prepend/append / or 0 for values in D/, DD, M/ or MM state.
+    const adjustDay = newValue => (newValue.endsWith('/') ? `0${newValue}` : `${newValue}/`);
+    const adjustMonth = newValue => {
+      const [day, month] = newValue.split('/');
+      return newValue.endsWith('/') ? `${day}/0${month}/` : `${day}/${month}/`;
+    };
     const testString = newValue => regexpLookup[newLength].test(newValue);
 
-    if (newLength > 7) return null;
+    if (newLength > 10) return null;
     if (newLength === 0) return setDate('');
     if (newLength < oldLength) return testString(value) ? setDate(value) : setDate('');
-    if (newLength === 2) return testString(value) ? setDate(adjustMonth(value)) : setDate(date);
+    if (newLength === 2) return testString(value) ? setDate(adjustDay(value)) : setDate(date);
+    if (newLength === 5) return testString(value) ? setDate(adjustMonth(value)) : setDate(date);
     return testString(value) ? setDate(value) : setDate(date);
   };
 
-  // Finalises an expiry date to be in the format MM/YYYY if the user has entered
-  // MM/Y MM/YY MM/YYY by placing a potentially incorrect year in the 20th century.
+  // Finalises an expiry date to be in the format DD/MM/YYYY.
   const finaliseDate = () => {
-    // If the expiry date isn't in at least the form MM/Y - reset.
-    if (date.length < 4) return setDate('');
+    // If the expiry date isn't in at least the form DD/MM/Y - reset.
+    if (date.length < 7) return setDate('');
 
-    const [month, year] = date.split('/');
+    const [day, month, year] = date.split('/');
     const prependedYear = `${'200'.substring(0, 4 - year.length)}${year}`;
-    return setDate(`${month}/${prependedYear}`);
+
+    return setDate(`${day}/${month}/${prependedYear}`);
```

Comment:
This function is predicated on `date` never being an invalid string, previously relying on the regex to ensure that. Since it's not ensuring that anymore, it's not really valid.

If you do type `"31/02/2020"` then un-focus the cell, the actual text within the cell is not reset, as this is a stateful component and the current state is the `date` string. The value that actually gets set in the realm is `null`, though. Easiest solution would probs be to just chuck a `moment(..).isValid()` in thiis `finaliseDate` which would `setDate('')` if false - that should keep things consistent